fluster/test.py

```diff
@@ -94,14 +95,71 @@ def _test(self) -> None:
             os.remove(output_filepath)
 
         if not self.reference:
-            self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.FAIL
-            if self.test_vector.result.lower() == result.lower():
-                self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.SUCCESS
-            self.assertEqual(
-                self.test_vector.result.lower(),
-                result.lower(),
-                self.test_vector.name,
-            )
+            self.compare_result(result, output_filepath, input_filepath)
         else:
             self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.REFERENCE
             self.test_suite.test_vectors[self.test_vector.name].result = result
+
+    @abstractmethod
+    def compare_result(self, result: str, output_filepath: str, input_filepath: str) -> None:
+        """Compare the test result with the expected value"""
+
+
+class MD5ComparisonTest(Test):
+    """Test class for MD5 comparison"""
+
+    def compare_result(self, result: str, output_filepath: str, input_filepath: str) -> None:
+        self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.FAIL
+        if self.test_vector.result.lower() == result.lower():
+            self.test_suite.test_vectors[self.test_vector.name].test_result = TestVectorResult.SUCCESS
+        self.assertEqual(
+            self.test_vector.result.lower(),
+            result.lower(),
+            self.test_vector.name,
+        )
+
+
+class PixelComparisonTest(Test):
+    """Test class for pixel comparison"""
+
+    def __init__(
+        self,
+        decoder: Decoder,
+        test_suite: Any,
+        test_vector: TestVector,
+        skip: bool,
+        output_dir: str,
+        reference: bool,
+        timeout: int,
+        keep_files: bool,
+        verbose: bool,
+        reference_decoder: Optional[Decoder] = None,
+    ):
+        super().__init__(decoder, test_suite, test_vector, skip, output_dir, reference, timeout, keep_files, verbose)
+        self.reference_decoder = reference_decoder
+        self.keep_files = True
+
+    def compare_result(self, result: str, output_filepath: str, input_filepath: str) -> None:
+        if not self.reference_decoder:
+            raise ValueError(f"Reference decoder not provided for pixel comparison test: {self.test_vector.name}")
+
+        reference_yuv = self.reference_decoder.decode(
+            input_filepath,
+            output_filepath,
+            self.test_vector.output_format,
+            self.timeout,
+            self.verbose,
+            self.keep_files,
+        )
+
+        if self.decoder.name == self.reference_decoder.name:
```

Comment:
There is no point on supporting this use case, an test suite that uses the pixel method can't be run for the reference decoder, it shouldn't be created/listed.

Reply:
Around here https://github.com/fluendo/fluster/blob/master/fluster/fluster.py#L279